File: base/src/main/java/de/darkatra/patcher/model/Packet.java
package de.darkatra.patcher.model;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Packet {
	private final String src;
	private final String dest;
	private final Long packetSize;
	private final LocalDateTime dateTime;
	private final String checksum;
	private final boolean backupExisting;
}

File: update-builder/src/main/kotlin/de/darkatra/patcher/updatebuilder/Packet.kt
package de.darkatra.patcher.updatebuilder

import com.fasterxml.jackson.annotation.JsonIgnore
import java.nio.file.Path
import java.time.Instant

data class Packet(
    val src: String,
    val dest: String,
    val packetSize: Long,
    val compressedSize: Long,
    val dateTime: Instant,
    val checksum: String,
    val backupExisting: Boolean,
    val compression: Compression,
    val feature: Feature? = null,
    @JsonIgnore
    val gzipPath: Path? = null
)
